test/gtest/gatherv2.hpp

```diff
@@ -0,0 +1,207 @@
+struct GatherV2BwdTest : public ::testing::TestWithParam<GatherV2TestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle = get_handle();
+        config        = GetParam();
+
+        auto param_grad_dims = config.GetParamGradDim();
+        auto indices_dims    = config.GetIndicesDim();
+        dim                  = config.GetDim();
+        batch_dims           = config.GetBatchDims();
+
+        std::vector<size_t> out_grad_dims;
+        for(uint32_t i = 0; i < dim; i++)
+        {
+            out_grad_dims.push_back(param_grad_dims[i]);
+        }
+
+        for(uint32_t i = batch_dims; i < indices_dims.size(); i++)
+        {
+            out_grad_dims.push_back(indices_dims[i]);
+        }
+
+        for(uint32_t i = dim + 1; i < param_grad_dims.size(); i++)
+        {
+            out_grad_dims.push_back(param_grad_dims[i]);
+        }
+
+        gatherDesc.setMode(MIOPEN_GATHER_V2);
+        gatherDesc.setDim(dim);
+        gatherDesc.setBatchDims(batch_dims);
+
+        auto gen_value = [](auto...) { return prng::gen_descreet_uniform_sign<T>(1e-2, 100); };
+        auto gather_dim_size = param_grad_dims[dim];
+        auto gen_index       = [gather_dim_size](auto...) {
+            return prng::gen_0_to_B(static_cast<I>(gather_dim_size));
+        };
+
+        indices    = tensor<I>{indices_dims}.generate(gen_index);
+        outputGrad = tensor<T>{out_grad_dims}.generate(gen_value);
+
+        paramGrad = tensor<T>{param_grad_dims};
+        std::fill(paramGrad.begin(), paramGrad.end(), static_cast<T>(0));
+
+        ref_paramGrad = tensor<T>{param_grad_dims};
+        std::fill(ref_paramGrad.begin(), ref_paramGrad.end(), static_cast<T>(0));
+
+        paramGrad_dev  = handle.Write(paramGrad.data);
+        indices_dev    = handle.Write(indices.data);
+        outputGrad_dev = handle.Write(outputGrad.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        cpu_gatherv2_backward<T, I>(outputGrad, indices, ref_paramGrad, dim, batch_dims);
+
+        miopenStatus_t status;
+        status = gatherDesc.Backward(handle,
+                                     outputGrad.desc,
+                                     outputGrad_dev.get(),
+                                     indices.desc,
+                                     indices_dev.get(),
+                                     paramGrad.desc,
+                                     paramGrad_dev.get());
+
+        EXPECT_EQ(status, miopenStatusSuccess);
+
+        paramGrad.data = handle.Read<T>(paramGrad_dev, paramGrad.data.size());
+    }
+
+    double GetTolerance()
+    {
+        double tolerance = std::numeric_limits<T>::epsilon() * 10;
+        return tolerance;
+    }
+
+    void Verify()
+    {
+        double threshold = GetTolerance();
+        auto error       = miopen::rms_range(ref_paramGrad, paramGrad);
+
+        ASSERT_EQ(miopen::range_distance(ref_paramGrad), miopen::range_distance(paramGrad));
```

Comment:
`range_distance` equality is already checked in function `rms_range()`, which is called in line 184. So this line is unnecessary

Reply:
nice finding, I've removed that